test/OSTPrime/allowance.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const chai = require('chai');
+const Web3 = require('web3');
+const sinon = require('sinon');
+
+const assert = chai.assert;
+const OSTPrime = require('../../src/ContractInteract/OSTPrime');
+const AssertAsync = require('../../test_utils/AssertAsync');
+const SpyAssert = require('../../test_utils/SpyAssert');
+const Utils = require('../../src/utils/Utils');
+
+describe('OSTPrime.allowance()', () => {
+  let web3;
+  let ostPrimeAddress;
+  let ostPrime;
+
+  beforeEach(() => {
+    web3 = new Web3();
+    ostPrimeAddress = '0x0000000000000000000000000000000000000002';
+    ostPrime = new OSTPrime(web3, ostPrimeAddress);
+  });
+
+  it('should pass with correct params', async () => {
+    let expectedAllowance = '100';
+    const spyAllowance = sinon.replace(
+      ostPrime.contract.methods,
+      'allowance',
+      sinon.fake.returns({
+        call: () => Promise.resolve(expectedAllowance),
+      }),
+    );
+
+    const spyMethod = sinon.spy(ostPrime, 'allowance');
```

Comment:
Here you would have a naming conflict. Maybe you can find better non-conflicting names.